app/controllers/hyrax/depositors_controller.rb

```diff
@@ -15,14 +14,18 @@ def index
     end
 
     def create
-      grantor = authorize_and_return_grantor
-      grantee = ::User.from_url_component(params[:grantee_id])
-      if grantor.can_receive_deposits_from.include?(grantee)
-        head :ok
+      if params[:grantee_id] == ::User.from_url_component(params[:user_id]).email
```

Comment:
Looking at this again, I wonder if it makes more sense (and maybe saves a little bit of processing in the normal case) to move this test into the if/else that starts on line 22.  Then it could be something like:
```
if grantor == grantee
  render_json_response(response_type: :unprocessable_entity, message: view_context.t('hyrax.dashboard.proxy_add_deny'))
elsif grantor.can_receive_deposits_from.include?(grantee)
...
```